Add how to for configuring QtCreator to use the SDK

kursatkobya opened the pull request:
Since the add-kit script in pelux-sde is not fullfilling build requirements yet (cant build cmake projects), here it is described how to configure QtCreator manually.

Please review and leave comments.

Signed-off-by: kursat <KKobya@luxoft.com>

kursatkobya committed:
Add how to for configuring QtCreator to use the SDK

Signed-off-by: kursat <KKobya@luxoft.com>

jeremiah commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+
+After the installation of the SDK and sourcing environment, there are couple of
+steps to do for running QtCreator in target environment. This roadmap assumes
+that sdk and QtCreator should be installed and environments are sourced already.
```
I think consistency is important and would recommend always capitalizing the acronym "SDK."

jeremiah commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+Configuring QtCreator to use the SDK
+====================================
+
+After the installation of the SDK and sourcing environment, there are couple of
```
What is a "sourcing environment"? Is it to distinguish between the "target" env and the "host" env? If so, should the more traditional term "host" be used?

kursatkobya replied:
it is like 
``source /opt/pelux_sdk/environment-setup-core2-64-pelux-linux`` on target platform.
this step is already described in ``using-the-sdk-to-crosscompile.rst`` section.

In that article there is a session named ``Sourcing Environment`` so this was reference to that one. Surely i can change it if it is unappropriate/

rpannek replied:
I guess it should have been called: "Sourcing the environment"

rpannek replied:
Change it to "... and sourcing the environment,"

rpannek replied:
and it should be "there are a copple of"

jonte replied:
You may even say `source`ing the environment, so make it clear you are referring to a command.

kursatkobya replied:
sure i can change it here/ Do you think it should be also changed in using-the-sdk-to-crosscompile.rst ? 

then i can change it there and add it to this PR as well.

breakreturn replied:
For maintenance reasons I'm wondering if we should have a pattern in how-tos where we state "Prerequisites" at the top and just have a list of links to other pages. This would reduce the need to rewrite all these introductory paragraphs we have everywhere which are there to basically just hint at where you are in "some part of a path" kind of.

Like:

Prerequisites
* Install SDK <----- this would just be a link
* Sourcing the environment <----- this would just be a link

Should also mention in this context that the overall design idea here is to have "Instruction" type articles that construct a path to follow in order to achieve something which is larger than a single how-to but smaller than an abstract process. This Instruction would explain what the purpose is and then present a list of how-tos that would make up what needs to be done.

For example it could be that "Working with the image on a target" (or whatever) would be an Instruction that explains the overall purpose, what the reader will accomplish, and then proceed to point out distinct how-tos for e.g. configuring the host, downloading the platform, building an image, building and SDK, sourcing the env, flashing, etc.

The purpose is to avoid ripple effects "upwards" in the SWF texts and between articles of the same "type" as much as possible.

jonte replied:
@kursatkobya In general, I think we should handle issues found in "old" commits as separate fixes. If you want to fix the issue you mention in using-the-sdk-to-crosscompile.rst, I think it should be a separate PR (and absolutely a separate commit!), since it will be easier to find the origins of the fix afterwards if we do it like this.

jonte replied:
Also, if there is a section about setting up the environment already, maybe you can add a footnote linking to it, so that the user is made aware of this section in case this is the first time the user encounters `source`.

breakreturn replied:
@jonte What do you think about my suggestion above? It sort of collides with what you just wrote about footnotes I believe? My proposal would mean that there is no reason to write a nice introduction but rather align the reader in the context by providing a list of prerequisites.

Thinking about this, the list of prerequisites is also a maintenance issue and should perhaps be simplified to be a link to the Instruction instead?

jonte replied:
@breakreturn 
There are pros and cons of both approaches. I am not sure which one I prefer, but let me list my views of both:

With footnotes/references in-lined (in general, all through the text), we can make a text which is possible to read "front to back" in a nice manner. This means that someone can take the entire SWF and just read it. This is similar to how a book is usually structured (e.g. you can pick up a chapter in the middle of a book, and expect to be referred back to important concepts that you may have missed if you do so).

With a prerequisites section, we end up with something that is closer to a "knowledge base", in which there is a very clear relation between articles, but it may be a bit drier to read than a book, and may not be as "comfortable" to read "front to back".

Regarding introductory pieces for information already conveyed elsewhere; I agree that we should keep this to a minimum. I would have given a short footnote ("for more info on concept X, see Y") - but that's just my personal opinion.

In all honesty though, I didn't see your original comment, and what I commented to @kursatkobya was just how I would rather see the reference being made :)

I am fine with either way, but we should be consistent. @breakreturn you have the best overview of the SWF - please choose what you prefer and I think both ways will be fine.

jonte replied:
We can refer back to this thread for a reasoning of our choice later on.

kursatkobya replied:
I believe i can give the link with a different commit after correcting all grammar typos in this one. If this is okay i will keep this as is the i will add another commit with links and corrections on ``using-the-sdk-to-crosscompile.rst``

breakreturn replied:
@jonte My proposal was partially intended to remove the need of having in-lined references since there would be no text. It's very un-book-like in structure as you say, but so far everything points to maintenance being the biggest issue with the SWF. I'd rather have a dry text and slightly un-natural structure than wrong content. The maintenance complexity is where my bias comes from in these things.

Maybe we can re-write the intro text here to be very minimal and only reference the other articles, and leave the final design decision to after we've had more design discussions?

jeremiah commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* If you are working with Qmake then it should be added as well. You can learn
+qmake path with command ``which qmake``
+* Add ``CMAKE_SYSROOT:STRING=<path of sysroot>`` item to ``CMake Configuration``
+list. You can learn sysroot path by typing ``echo ${SDKTARGETSYSROOT}`` to console.
```
"learn" here might be better described as "discover".

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+====================================
+
+After the installation of the SDK and sourcing environment, there are couple of
+steps to do for running QtCreator in target environment. This roadmap assumes
```
"... to do to run QtCreator with the target environment."

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+====================================
+
+After the installation of the SDK and sourcing environment, there are couple of
+steps to do for running QtCreator in target environment. This roadmap assumes
```
This seems incorrect - "running QtCreator in target environment" seems to imply you would run QtCreator on, for instance, a TX2 or qemu INSIDE PELUX - this isn't what you mean, I'm sure.

kursatkobya replied:
Yeah that's what i should write in SDK environment i guess.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+
+After the installation of the SDK and sourcing environment, there are couple of
+steps to do for running QtCreator in target environment. This roadmap assumes
+that SDK and QtCreator should be installed and environments are sourced already.
```
"that the SDK and QtCreator are installed and the environment is sourced already."

jonte replied:
`source`d, for clarity.

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+that SDK and QtCreator should be installed and environments are sourced already.
+
+Adding a new kit
+--------------------
```
Can we have screenshots to support this section? It will make it easier for users to understand what to do.

kursatkobya replied:
I can create and find a way to add them. good idea.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+Adding a new kit
+--------------------
+
+To be able to cross-compile via QtCreator, new build kit should be added to
```
"... Qtcreator, a new ..."  "... be added in QtCreator"

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+left panel of option window.
+* Select ``Kits`` tab and click ``Add`` button.
+* Name your new kit and choose ``Device type`` as ``GenericLinuxOsType``.
+* Then browse and submit the sysroot. If you don't know sysroot location you can
```
Make sure to note that this will only work in the console where the SDK environment script was `source`d.

rpannek replied:
"Then browse and select the path to the sysroot. If you don't know the sysroot location you can"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+QtCreator.
+
+* Open QtCreator
+* Open Options form from Tools->Options, then click ``Build & Run`` item on the
```
"click ``Build & Run`` in the"

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Then choose C compiler. If cross compiler of the SDK is not listed, it should
+be added as a new gcc. To add new gcc click ``Manage`` button next to compiler.
+Add you GCC compiler path here. You can discover this path from environment as well.
+Just type ``which ${CC}`` to console to discover.
```
Again, only in the same shell as you `source`d the environment in.

rpannek replied:
"To do so, type ``which ${CC}`` in your terminal."

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+Add you GCC compiler path here. You can discover this path from environment as well.
+Just type ``which ${CC}`` to console to discover.
+* Choose C++ compiler as you did in previous step. You can discover path by typing
+``which ${CXX}`` to console.
```
Same as above.

rpannek replied:
`which ${CXX}`` in the terminal.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+
+* Open QtCreator
+* Open Options form from Tools->Options, then click ``Build & Run`` item on the
+left panel of option window.
```
"of the option"

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Choose C++ compiler as you did in previous step. You can discover path by typing
+``which ${CXX}`` to console.
+* Copy the environment list of the SDK to ``Environment``. You can discover environment
+list by typing ``env`` to console.
```
Same as above.

rpannek replied:
"```env`` in the terminal"

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Copy the environment list of the SDK to ``Environment``. You can discover environment
+list by typing ``env`` to console.
+* Choose ``Debugger``, if it is not listed in dropdown menu click ``Manage..``
+and add a new Debugger. You can discover debugger path with command ``which $GDB``
```
Same as above.

rpannek replied:
"You can find the debugger path with ``which $GDB``"

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Choose ``Debugger``, if it is not listed in dropdown menu click ``Manage..``
+and add a new Debugger. You can discover debugger path with command ``which $GDB``
+* Choose CMake Tool, if it is not listed in dropdown menu click ``Manage..``
+button and add a new CMake. You can discover Cmake path with command : ``which cmake``
```
Cmake -> CMake

jonte replied:
+ same as above re: environment.

kursatkobya replied:
The command is cmake not CMake, so i suppoose i shoudl keep which cmake

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Open QtCreator
+* Open Options form from Tools->Options, then click ``Build & Run`` item on the
+left panel of option window.
+* Select ``Kits`` tab and click ``Add`` button.
```
"Select the ``Kits`` tab and click the ``Add`` button."

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+and add a new Debugger. You can discover debugger path with command ``which $GDB``
+* Choose CMake Tool, if it is not listed in dropdown menu click ``Manage..``
+button and add a new CMake. You can discover Cmake path with command : ``which cmake``
+* If you are working with Qmake then it should be added as well. You can discover
```
Qmake -> qmake

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Choose CMake Tool, if it is not listed in dropdown menu click ``Manage..``
+button and add a new CMake. You can discover Cmake path with command : ``which cmake``
+* If you are working with Qmake then it should be added as well. You can discover
+qmake path with command ``which qmake``
```
Same as above re: environment.

rpannek replied:
"the path to qmake with ``which qmake``"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Open Options form from Tools->Options, then click ``Build & Run`` item on the
+left panel of option window.
+* Select ``Kits`` tab and click ``Add`` button.
+* Name your new kit and choose ``Device type`` as ``GenericLinuxOsType``.
```
"as ``Device type`` choose ``GenericLinuxOsType``."

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* If you are working with Qmake then it should be added as well. You can discover
+qmake path with command ``which qmake``
+* Add ``CMAKE_SYSROOT:STRING=<path of sysroot>`` item to ``CMake Configuration``
+list. You can discover sysroot path by typing ``echo ${SDKTARGETSYSROOT}`` to console.
```
Same as above re: environment. I realize that I made a lot of comments about the environment, and that adding all of these to the text will look weird. Maybe add a very visible warning saying that the environment to observe is the one set up by the environment script.

rpannek replied:
"You can find the path to the sysroot with ``echo ...``."

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+
+
+How to open the template service project with QtCreator
+-------------------------------------------------------
```
Screenshots here too would be nice.

jonte commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Select ``Import Project`` from ``Projects`` section
+* Click on ``Git Clone`` and then ``Choose`` button
+* On the opening window add repository to ``Repository section``. The git repo
+for template-library is https://github.com/Pelagicore/template-library
```
template-library -> `template-library`

rpannek replied:
"for the .."

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Select ``Kits`` tab and click ``Add`` button.
+* Name your new kit and choose ``Device type`` as ``GenericLinuxOsType``.
+* Then browse and submit the sysroot. If you don't know sysroot location you can
+discover by typing ``echo ${SDKTARGETSYSROOT}`` to console.
```
"find it by runnung ``echo ${SDKTARGETSYSROOT}`` in your terminal."

kursatkobya replied:
this is kinda contradicts with Jeremiah's comment. For now i am using "you can discover by typing ``echo ${SDKTARGETSYSROOT}`` to console"

should i change it like you mentioned here ?

rpannek replied:
There are two things here, console, which I don't like but migt be ok and the word discover which me and Jeremiah think doesn't fit here. Be consistent and either use learn or find instead of discover in the whole article.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Name your new kit and choose ``Device type`` as ``GenericLinuxOsType``.
+* Then browse and submit the sysroot. If you don't know sysroot location you can
+discover by typing ``echo ${SDKTARGETSYSROOT}`` to console.
+* Then choose C compiler. If cross compiler of the SDK is not listed, it should
```
"Then choose the C compiler. If the cross compiler"

jonte reviewed (changes requested):
Looks great!

I have some small requests about making it clearer that the environment to observe, and import into QtCreator is the environment where the environment script was sourced.

Additionally, it would be great to have some screenshots in place. When you create these screenshots, please shop around for a good tool to capture and annotate screenshots that we can use for future screenshots as well.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Then browse and submit the sysroot. If you don't know sysroot location you can
+discover by typing ``echo ${SDKTARGETSYSROOT}`` to console.
+* Then choose C compiler. If cross compiler of the SDK is not listed, it should
+be added as a new gcc. To add new gcc click ``Manage`` button next to compiler.
```
"To add a new gcc click the ``Manage`` button next to the compiler"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+discover by typing ``echo ${SDKTARGETSYSROOT}`` to console.
+* Then choose C compiler. If cross compiler of the SDK is not listed, it should
+be added as a new gcc. To add new gcc click ``Manage`` button next to compiler.
+Add you GCC compiler path here. You can discover this path from environment as well.
```
"Add your GCC", "path from the environment"

kursatkobya replied:
Since we are saying "you can discover this path from environment as well" afterwards, "Add your GCC path here" would be more appropriate i guess.

breakreturn reviewed (changes requested):
I have one comment of the more abstract kind which I think should be considered, Apart from that it looks good but I want the "how-to" tag to be used if this is a how-to article.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+be added as a new gcc. To add new gcc click ``Manage`` button next to compiler.
+Add you GCC compiler path here. You can discover this path from environment as well.
+Just type ``which ${CC}`` to console to discover.
+* Choose C++ compiler as you did in previous step. You can discover path by typing
```
"Choose the C++ compiler as you did the the previous step. You can find the path running"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+Just type ``which ${CC}`` to console to discover.
+* Choose C++ compiler as you did in previous step. You can discover path by typing
+``which ${CXX}`` to console.
+* Copy the environment list of the SDK to ``Environment``. You can discover environment
```
"You can find the environment"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+``which ${CXX}`` to console.
+* Copy the environment list of the SDK to ``Environment``. You can discover environment
+list by typing ``env`` to console.
+* Choose ``Debugger``, if it is not listed in dropdown menu click ``Manage..``
```
"in the dropdown"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+list by typing ``env`` to console.
+* Choose ``Debugger``, if it is not listed in dropdown menu click ``Manage..``
+and add a new Debugger. You can discover debugger path with command ``which $GDB``
+* Choose CMake Tool, if it is not listed in dropdown menu click ``Manage..``
```
"click the ``Manage``"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Choose ``Debugger``, if it is not listed in dropdown menu click ``Manage..``
+and add a new Debugger. You can discover debugger path with command ``which $GDB``
+* Choose CMake Tool, if it is not listed in dropdown menu click ``Manage..``
+button and add a new CMake. You can discover Cmake path with command : ``which cmake``
```
"add the path to CMake. You can find the path to CMake with ``which cmake``"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+button and add a new CMake. You can discover Cmake path with command : ``which cmake``
+* If you are working with Qmake then it should be added as well. You can discover
+qmake path with command ``which qmake``
+* Add ``CMAKE_SYSROOT:STRING=<path of sysroot>`` item to ``CMake Configuration``
```
"Add the ..."

rpannek replied:
"item to the ``CMake Configuration``"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+-------------------------------------------------------
+
+* Open QtCreator
+* From ``File` menu choose ``New File or Project``
```
"From the ``File`` menu choose ..."

You missed the "the" and the closing ` with File.

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+
+* Open QtCreator
+* From ``File` menu choose ``New File or Project``
+* Select ``Import Project`` from ``Projects`` section
```
"from the"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Open QtCreator
+* From ``File` menu choose ``New File or Project``
+* Select ``Import Project`` from ``Projects`` section
+* Click on ``Git Clone`` and then ``Choose`` button
```
Click ``Git Clone`` and then click the ``Choose`` button

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* From ``File` menu choose ``New File or Project``
+* Select ``Import Project`` from ``Projects`` section
+* Click on ``Git Clone`` and then ``Choose`` button
+* On the opening window add repository to ``Repository section``. The git repo
```
"In the window, which just opened, add the repository to the ..."

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* On the opening window add repository to ``Repository section``. The git repo
+for template-library is https://github.com/Pelagicore/template-library
+* Select a proper path and add it to ``Path`` section then click ``Next``
+* QtCreator will clone the project then click ``Finish``
```
"the project, then click"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+for template-library is https://github.com/Pelagicore/template-library
+* Select a proper path and add it to ``Path`` section then click ``Next``
+* QtCreator will clone the project then click ``Finish``
+* Here you should select the kit to use for building project
```
"kit to use to build your project"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,52 @@
+* Select a proper path and add it to ``Path`` section then click ``Next``
+* QtCreator will clone the project then click ``Finish``
+* Here you should select the kit to use for building project
+* Select the kit you have already created in previous step and click ``Configure``
```
"in the previous"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+Configuring QtCreator to use the SDK
+====================================
+
+After the installation of the SDK and ``source``ing the environment, there are couple of
```
"are a couple of"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+
+After the installation of the SDK and ``source``ing the environment, there are couple of
+steps to do to run QtCreator in the SDK environment. This roadmap assumes
+that SDK and QtCreator are installed and the environment is ``source``d already.
```
"the SDK"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+Adding a new kit
+--------------------
+
+To be able to cross-compile via QtCreator, a new build kit should be added to
```
You don't really add a kit to QtCreator, only to this setup. Just rephase it to "should be added."

kursatkobya replied:
Not sure if i get it correctly but done some change

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+* Select the ``Kits`` tab and click the ``Add`` button.
+* Name your new kit and as ``Device type`` choose ``GenericLinuxOsType``.
+* Then browse and select the path to the sysroot. If you don't know the sysroot
+location you can discover by typing ``echo ${SDKTARGETSYSROOT}`` to the terminal.
```
"discover" is not appropriate here, use "find it".

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+* Then browse and select the path to the sysroot. If you don't know the sysroot
+location you can discover by typing ``echo ${SDKTARGETSYSROOT}`` to the terminal.
+Please note that the terminal should be ``source``d beforehand.
+* Then choose C compiler. If the cross compiler of the SDK is not listed, it should
```
"choose a C compiler"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+location you can discover by typing ``echo ${SDKTARGETSYSROOT}`` to the terminal.
+Please note that the terminal should be ``source``d beforehand.
+* Then choose C compiler. If the cross compiler of the SDK is not listed, it should
+be added as a new gcc. To add new gcc click the ``Manage`` button next to compiler.
```
"To add a new gcc"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+Please note that the terminal should be ``source``d beforehand.
+* Then choose C compiler. If the cross compiler of the SDK is not listed, it should
+be added as a new gcc. To add new gcc click the ``Manage`` button next to compiler.
+Add your GCC path here. You can discover this path from environment as well.
```
"discover" is not appropriate here, use "find this path in the environment"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+Add your GCC path here. You can discover this path from environment as well.
+To do so, type ``which ${CC}`` in your terminal. Please note that the terminal
+should be ``source``d beforehand.
+* Choose the C++ compiler as you did in the previous step. You can discover path by
```
don't use discover use "You can find the path by"

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+Please note that the terminal should be ``source``d beforehand.
+* Choose CMake Tool, if it is not listed in the dropdown menu click the ``Manage..``
+button and add the path to CMake. You can find the path to Cmake with ``which cmake``.
+Please note that the terminal should be ``source``d beforehand.
```
Remove all the "Please note that the terminal should be ``source``d beforehand." from each line and add it at the top as "Please note that the terminal should be ``source``d beforehand for the following actions:"

kursatkobya replied:
This creates a conflict to @jonte 's request.

jonte replied:
@kursatkobya No.. What I meant with my original request was simply to highlight that these commands should be run in the right environment - I didn't mean that you needed to highlight this in each individual sentence :)

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+* Choose CMake Tool, if it is not listed in the dropdown menu click the ``Manage..``
+button and add the path to CMake. You can find the path to Cmake with ``which cmake``.
+Please note that the terminal should be ``source``d beforehand.
+* If you are working with qmake then it should be added as well. You can discover
```
don't use discover, use "find" instead

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+* If you are working with qmake then it should be added as well. You can discover
+the path to qmake with ``which qmake``. Please note that the terminal should be
+``source``d beforehand.
+* Add ``CMAKE_SYSROOT:STRING=<path of sysroot>`` item to the ``CMake Configuration``
```
"Add the ..."
\<path of sysroot\> should be \<sysroot path\> or \<path to sysroot\>

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+the path to qmake with ``which qmake``. Please note that the terminal should be
+``source``d beforehand.
+* Add ``CMAKE_SYSROOT:STRING=<path of sysroot>`` item to the ``CMake Configuration``
+list. You can find the path to the sysroot with ``echo ${SDKTARGETSYSROOT}`` to the terminal.
```
remove "to the terminal" here.

jonte reviewed (approved):
Looks good. Looking forward to screenshots as well :)

rpannek commented on docs/articles/sdk/configure-qtcreator.rst:
```diff
@@ -0,0 +1,59 @@
+* Click ``Git Clone`` and then click the ``Choose`` button
+* In the window, which just opened, add the repository to the ``Repository section``.
+The git repo for the ``template-library`` is https://github.com/Pelagicore/template-library
+* Select a proper path and add it to ``Path`` section then click ``Next``
```
"add it to the ``Path`` section, then" (don't miss the ",")

DunderRoffe reviewed (changes requested):
Are you sure this renders correctly?
When viewing in github the "Add kit" list only have 2 bullets and the rest renders as a text block.

I would guess that you need to do some indentation after line breaks in a bullet

kursatkobya committed:
Add screenshots to QtCreator configuration howto

Signed-off-by: kursat <KKobya@luxoft.com>

kursatkobya committed:
Add referral link to "Sourcing The Environment" documentation

After creating documentation for QtCreator configuration Howto, a
necessity arised to link reference how=to documentations. Thus, a
referral link is added to
docs/articles/sdk/using-the-sdk-to-crosscompile.rst and it is used in
docs/articles/sdk/configure-qtcreator.rst

Signed-off-by: kursat <KKobya@luxoft.com>

breakreturn commented on docs/articles/sdk/using-the-sdk-to-crosscompile.rst:
```diff
@@ -6,7 +6,9 @@ Using the SDK to cross compile
 some more pitfalls, but let's get on with the happy cases for now.
 
-Sourcing Environment
+.. _sourcing-environment:
```
I think this should be renamed as well to follow the same pattern of reflecting the heading.

breakreturn reviewed (changes requested):
Small comment
